controller/tests/array_action/svc/array_mediator_svc_test.py

```diff
@@ -591,29 +622,68 @@ def test_create_snapshot_as_stretched_success(self):
 
         self.svc.create_snapshot("source_volume_id", "test_snapshot", space_efficiency=None, pool=None)
         self.svc.client.svctask.mkvolume.assert_called_once_with(name='test_snapshot', unit='b', size=1024,
                                                                  pool='pool_name', iogrp='iogrp0',
                                                                  compressed=True, deduplicated=True)
 
     def test_create_snapshot_with_different_space_efficiency_success(self):
-        self._prepare_mocks_for_create_snapshot(source_has_deduplicated_copy=True)
+        self._prepare_mocks_for_create_snapshot_mkvolume(source_has_deduplicated_copy=True)
 
         self.svc.create_snapshot("source_volume_id", "test_snapshot", space_efficiency="thin", pool=None)
         self.svc.client.svctask.mkvolume.assert_called_once_with(name='test_snapshot', unit='b', size=1024,
                                                                  pool='pool_name', iogrp='iogrp0', thin=True)
 
     def test_create_snapshot_no_deduplicated_copy_success(self):
-        self._prepare_mocks_for_create_snapshot(support_deduplicated_copy=False)
+        self._prepare_mocks_for_create_snapshot_mkvolume(support_deduplicated_copy=False)
 
         snapshot = self.svc.create_snapshot("source_volume_id", "test_snapshot", space_efficiency=None, pool="pool1")
 
         self.assertEqual(snapshot.capacity_bytes, 1024)
         self.assertEqual(snapshot.array_type, 'SVC')
         self.assertEqual(snapshot.id, 'snap_id')
 
+    def _prepare_mocks_for_create_snapshot_addsnapshot(self):
+        self.svc.client.svctask.addsnapshot = Mock()
+        source_volume_to_copy_from = self._get_custom_cli_volume(False, False, pool_name='pool1')
+        volumes_to_return = [source_volume_to_copy_from, source_volume_to_copy_from]
+        self.svc.client.svcinfo.lsvdisk.side_effect = self._mock_cli_objects(volumes_to_return)
+        self.svc.client.svctask.addsnapshot.return_value = Mock(
+            response=(b'Snapshot, id [0], successfully created or triggered\n', b''))
+        self.svc.client.svcinfo.lsvolumesnapshot = Mock()
+        self.svc.client.svcinfo.lsvolumesnapshot.return_value = self._mock_cli_object(self._get_cli_snapshot())
+
+    def test_create_snapshot_addsnapshot_success(self):
+        self._prepare_mocks_for_create_snapshot_addsnapshot()
+        snapshot = self.svc.create_snapshot("source_volume_id", "test_snapshot", space_efficiency=None, pool="pool1")
+
+        self.assertEqual(snapshot.capacity_bytes, 1024)
+        self.svc.client.svctask.addsnapshot.assert_called_once_with(name='test_snapshot', volumes='test_id',
+                                                                    pool='pool1')
+        self.svc.client.svcinfo.lsvolumesnapshot.assert_called_once_with(object_id=0)
+        self.assertEqual(snapshot.array_type, 'SVC')
+        self.assertEqual(snapshot.id, '')
+        self.assertEqual(snapshot.internal_id, 'snapshot_id')
```

Comment:
as per our [conversation](https://stackoverflow.com/a/57059952) - do we want to use it correctly in this PR?
or should we refactor throughout the project in a separate ticket?

Reply:
lets do it after #501 is merged

Reply:
good idea

Reply:
done